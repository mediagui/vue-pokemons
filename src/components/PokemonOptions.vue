<template>
  <div class="options-container">
    <ul>
      <!-- 
        Recorremos el array de pokemons y pintamos su nombre.

        Al hacer click emitimos un evento personalizado (selection-pokemon) 
        y enviamos el id del pokemon seleccionado.
      -->
      <li
        v-for="pokemon in pokemons"
        :key="pokemon.id"
        @click="$emit('selection-pokemon', pokemon.id)"
      >
        {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    // Array de pokemons. Propiedad en la que recibimos el array de los pokemons seleccionado al azar
    pokemons: {
      type: Array,
      required: true,
    },
  },
};
</script>

<style scoped>
ul {
  list-style-type: none;
}

li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  padding: 5px 0;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>