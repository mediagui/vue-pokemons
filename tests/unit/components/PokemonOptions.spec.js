import { shallowMount } from "@vue/test-utils";
import  PokemonOptions from "@/components/PokemonOptions";


import { pokemons } from "../mocks/pokemons.mock";

describe('PokemonOptions Componente', () => {

    let wrapper


    beforeEach( () => {

        wrapper = shallowMount(PokemonOptions, {
            props:{
                pokemons // Equivalente a  pokemons: pokemons
            }
        })        

    })

    test('debe hacer match con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })


    test('debe de mostrar las 4 opciones correctamente', () => {
    
        const liTags = wrapper.findAll('li')
        expect(liTags.length).toBe(4)

        expect(liTags[0].text()).toBe('pikachu')
        expect(liTags[1].text()).toBe('charmander')
        expect(liTags[2].text()).toBe('venusaur')
        expect(liTags[3].text()).toBe('mew')
        
    })


    test('debe emitir "selection-pokemon" con sus respectivos parÃ¡metros al hacer click' , () => {
    
        // desestructuramos los elementos li en los que tenemos los pokemons
        const [li1, li2, li3, li4] = wrapper.findAll('li')

        //click manual sobre los elementos
        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        expect( wrapper.emitted('selection-pokemon').length ).toBe(4) // Por lo menos emitido 1 vez
        expect( wrapper.emitted('selection-pokemon')[0] ).toStrictEqual([5]) // Se emite el evento con un arreglo de un elemento con el valor 5
        expect( wrapper.emitted('selection-pokemon')[1] ).toStrictEqual([10]) // Se emite el evento con un arreglo de un elemento con el valor 10
        expect( wrapper.emitted('selection-pokemon')[2] ).toStrictEqual([15]) // Se emite el evento con un arreglo de un elemento con el valor 15
        expect( wrapper.emitted('selection-pokemon')[3] ).toStrictEqual([20]) // Se emite el evento con un arreglo de un elemento con el valor 20

    })

})

