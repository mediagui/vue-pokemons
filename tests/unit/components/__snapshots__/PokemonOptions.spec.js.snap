// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Componente debe hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <!-- 
        Recorremos el array de pokemons y pintamos su nombre.

        Al hacer click emitimos un evento personalizado (selection-pokemon) 
        y enviamos el id del pokemon seleccionado.
      -->
    <li>pikachu</li>
    <li>charmander</li>
    <li>venusaur</li>
    <li>mew</li>
  </ul>
</div>
`;
